docs-src/java/how-to-use-a-custom-payload-codec-in-java.md

```diff
@@ -0,0 +1,111 @@
+---
+id: how-to-use-a-custom-payload-codec-in-java
+title: How to use a custom Payload Codec in Java
+sidebar_label: Custom Payload Codec
+description: Create a custom implementation of `PayloadCodec` and use it in `CodecDataConverter` to set a custom Data Converter.
+tags:
+  - Java
+  - developer-guide
+---
+
+**Create a custom Payload Codec**
+
+Create a custom implementation of [`PayloadCodec`](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/payload/codec/PayloadCodec.html) and use it in [`CodecDataConverter`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/CodecDataConverter.html)) to set a custom Data Converter.
```

Comment:
We need to unify the style of the links to reference docs.
I'm not sure what is our style.
If it's "latest", all links should be not to tags, but to the "latest" state.
If we reference some specific tag (which has it's own merit that such links will never die), it should be tag-based urls for all languages on all pages.

Reply:
yes; I'm checking all links and correcting them to point to latest.